src/test/java/dev/aherscu/instancio/experiment/PersonRequestHandlerTest.java

```diff
@@ -1,58 +1,77 @@
 
+    private static Model<CreatePersonRequest> createValidRequestModel() {
+        return Instancio.of(CreatePersonRequest.class)
+                .withSettings(Settings.create()
+                        .set(Keys.BEAN_VALIDATION_ENABLED, true))
+                .toModel();
+    }
+
     @DataProvider
     private static CreatePersonRequest[] data() {
-        return new CreatePersonRequest[] {
-            CreatePersonRequest.builder()
-                .person(Person.builder()
-                    .address(Address.builder()
-                        .line1("1")
-                        .city("A")
-                        .country("A")
-                        .zipCode("1")
-                        .build())
-                    .phone(Phone.builder()
-                        .countryCode(1)
-                        .number(1)
-                        .build())
-                    .build())
-                .build(),
-        };
+        return Instancio.stream(createValidRequestModel())
+                .limit(100)
+                .toArray(CreatePersonRequest[]::new);
     }
 
     @DataProvider
     private static CreatePersonRequest[] negativeData() {
-        return new CreatePersonRequest[] {
-            CreatePersonRequest.builder()
-                .build(),
-            CreatePersonRequest.builder()
-                .person(Person.builder()
-                    .build())
-                .build(),
-            CreatePersonRequest.builder()
-                .person(Person.builder()
-                    .address(Address.builder()
-                        .line1("1")
-                        .city("A")
-                        .country("A")
-                        .zipCode("1")
-                        .build())
-                    .phone(Phone.builder()
-                        .countryCode(10000)
-                        .number(1)
-                        .build())
-                    .build())
-                .build(),
-        };
+        // either empty or a string of 21+ characters in length
+        final StringSpec emptyStringOrLengthOver21 = Gen.string().allowEmpty().minLength(21);
+        final Generator<?> nullGenerator = random -> null;
+
+        // create an otherwise valid object, but with a single invalid field
+        final List<Tuple<TargetSelector, GeneratorSpec<?>>> negativeInputs = Arrays.asList(
+            Tuple.of(field(Address::getLine1), Gen.string().nullable().allowEmpty().minLength(21)), // line1 is not nullable
+            Tuple.of(field(Address::getLine2), emptyStringOrLengthOver21),
+            Tuple.of(field(Address::getCity), emptyStringOrLengthOver21),
+            Tuple.of(field(Address::getCountry), emptyStringOrLengthOver21),
+            Tuple.of(field(Address::getZipCode), emptyStringOrLengthOver21),
+            // random number outside the range [1-999]
+            Tuple.of(field(Phone::getCountryCode), Gen.ints().range(Integer.MIN_VALUE, 0).range(1000, Integer.MAX_VALUE)),
```

Comment:
This also means that whenever the model changes, e.g. a new class or field add/delete/chaged, these rules have to updated here, all in one place.

Reply:
That's right.